test/cppsuite/tests/run.cxx

```diff
@@ -37,6 +37,14 @@
 #include "example_test.cxx"
 #include "hs_cleanup.cxx"
 
+namespace test_harness {
+/*
+ * FIXME-WT-7782: This needs to exist in a single translation unit. At the moment, almost everything
+ * is in headers so the only reasonable spot to put it is here.
+ */
+std::mutex _logger_mtx;
```

Comment:
Why do we have to define the mutex outside of debug_utils.h?

Reply:
We need to define it here otherwise we would run into `undefined reference` during compilation. When we will do the work defined in WT-7782, we should be able to move it to `debug_utils.cxx`.

Reply:
When I say here... It could be any translation unit that includes `"test_harness/util/debug_utils.h"`

Reply:
Sorry, my bad

Reply:
Yeah, the idea is that it should be defined in **one** TU. And whenever another TU uses it, it should be using the extern declaration. That tells the compiler that it doesn't need to find the full definition and that it should defer finding it to the linking phase (which is what you want, it's global so they should all be using the same memory). If you don't do this, you get a copy in each TU that brings in the header.